File: src/main/java/com/hileco/cortex/server/parsing/ExpressionSerializer.java
package com.hileco.cortex.server.parsing;

import com.fasterxml.jackson.core.JsonGenerator;
import com.fasterxml.jackson.databind.SerializerProvider;
import com.fasterxml.jackson.databind.ser.std.StdScalarSerializer;
import com.fasterxml.jackson.databind.ser.std.StringSerializer;
import com.hileco.cortex.constraints.expressions.Expression;

import java.io.IOException;

public class ExpressionSerializer extends StdScalarSerializer<Expression> {

    private final StringSerializer stringSerializer;

    public ExpressionSerializer() {
        super(Expression.class);
        this.stringSerializer = new StringSerializer();
    }

    @Override
    public void serialize(Expression value, JsonGenerator gen, SerializerProvider provider) throws IOException {
        this.stringSerializer.serialize(value == null ? null : value.toString(), gen, provider);
    }
}

File: src/main/java/com/hileco/cortex/server/serialization/ExpressionSerializer.kt
package com.hileco.cortex.server.serialization

import com.fasterxml.jackson.core.JsonGenerator
import com.fasterxml.jackson.databind.SerializerProvider
import com.fasterxml.jackson.databind.ser.std.StdScalarSerializer
import com.fasterxml.jackson.databind.ser.std.StringSerializer
import com.hileco.cortex.constraints.expressions.Expression

class ExpressionSerializer : StdScalarSerializer<Expression>(Expression::class.java) {
    private val stringSerializer: StringSerializer = StringSerializer()

    override fun serialize(value: Expression?, gen: JsonGenerator, provider: SerializerProvider) {
        stringSerializer.serialize(value?.toString(), gen, provider)
    }
}
